compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -141,6 +139,28 @@ class ExpressionCodegen(
         mv.visitEnd()
     }
 
+    private fun markFunctionLineNumber() {
+        if (irFunction.origin == JvmLoweredDeclarationOrigin.CLASS_STATIC_INITIALIZER) {
+            return
+        }
+        if (irFunction is IrConstructor && irFunction.isPrimary) {
```

Comment:
This method called after function body generation (irFunction.body!!.accept(this, info)). Should we use  markLineNumber(startOffset = FALSE)?

Reply:
I tried to match the current behavior, but was not able to do so precisely since some code is not exactly relevant to IR (see for example markLineNumberForConstructor). As a result I decided to make sure we match what is reflected in tests and them fix the rest of the cases when we find them.

In line #146, for example, primary constructor end offset location location is taken from the class at least in some cases, which leads to weird breakpoints.